### PR TITLE
meson: Add missing build options to enable building on macOS

### PR DESCRIPTION
**Pull request by jwillikers**
Sorry for the large PR, I can break into smaller PR's as necessary. I tried to keep the commits as small and focused as possible.

These are the remaining build options and tweaks to get Meson building on macOS, at least when using `meson setup -Dauto_features=disabled ...`. Still, macOS builds will continue to be blocked by issue #2873.

I should be able to follow up with additional fixes for when `-Dauto_features=disabled` is omitted.

### CODE REVIEW

**Comment by karelzak** [comment]
@t-8ch please, review ;-)

File: meson.build [hunk 1]
```diff
@@ -1575,105 +1578,135 @@ exes += exe
 manadocs += ['sys-utils/ctrlaltdel.8.adoc']
 bashcompletions += ['ctrlaltdel']
 
+opt = get_option('build-fsfreeze').require(cc.has_header('linux/fs.h')).allowed()
```
Comment (t-8ch) [inline meson.build:1581]: Why not fetch this from `conf` where it has already been checked?

**Review by t-8ch — APPROVED** [review]
One tiny question, otherwise looks good.

File: meson.build (hunk 1 above)
Comment (jwillikers) [reply meson.build:1581]: Oops, my bad. I'll do that.

Comment (jwillikers) [reply meson.build]: Done.

**Pull request merged**